arch/x86/core/ia32/crt0.S

```diff
@@ -152,43 +152,6 @@ __csSet:
 	addl	$CONFIG_ISR_STACK_SIZE, %esp
 #endif
 
-#ifdef CONFIG_XIP
-	/*
```

Comment:
Does this break XIP, or do we not support that with x86 anymore?  Worth a comment explaining at least.

Reply:
We still have `qemu_x86_xip` to test XIP, and tests are still passing in CI. Note that the copying of data is still being done at the beginning or `z_x86_prep_c()` by calling `z_data_copy()`.

Reply:
Found that it actually broke with `CONFIG_GDT_DYNAMIC=y` even though CI thinks everything is fine. Fixed it by moving things around.